hack/benchmark/time-to-k8s/public-chart/generate-chart.go

```diff
@@ -0,0 +1,211 @@
+package main
+
+import (
+	"encoding/csv"
+	"encoding/json"
+	"flag"
+	"image/color"
+	"io"
+	"log"
+	"os"
+	"strconv"
+	"time"
+
+	"gonum.org/v1/plot"
+	"gonum.org/v1/plot/plotter"
+	"gonum.org/v1/plot/vg"
+	"gonum.org/v1/plot/vg/draw"
+)
+
+type data struct {
+	Date   time.Time `json:"date"`
+	Cmd    float64   `json:"cmd"`
+	Api    float64   `json:"api"`
+	K8s    float64   `json:"k8s"`
+	DnsSvc float64   `json:"dnsSvc"`
+	App    float64   `json:"app"`
+	DnsAns float64   `json:"dnsAns"`
+	Total  float64   `json:"total"`
+}
+
+type thing struct {
+	Data []data `json:"data"`
+}
+
+func main() {
+	csvPath := flag.String("csv", "", "path to the CSV file")
+	chartPath := flag.String("output", "", "path to output the chart to")
+	pastRunsPath := flag.String("past-runs", "", "path to the JSON contains past runs data")
+	flag.Parse()
+
+	qq := readInCSV(*csvPath)
+	d := readData(*pastRunsPath)
+	d.Data = append(d.Data, qq)
+	updateJSON(d, *pastRunsPath)
+	createChart(d, *chartPath)
+}
+
+func readInCSV(csvPath string) data {
```

Comment:
Change name for both func and the data structures and add comment

I would like to be able to read the code without looking at each data structures and implemenation of func
5to see what it doing.

Currently i have to read the func to see what it does
And then look at every data structure to see what it actually holds